pkg/scheduler/objects/queue.go

```diff
@@ -663,29 +663,61 @@ func (sq *Queue) AddApplication(app *Application) {
 	// YUNIKORN-199: update the quota from the namespace
 	// get the tag with the quota
 	quota := app.GetTag(common.AppTagNamespaceResourceQuota)
-	if quota == "" {
+	// get the tag with the guaranteed resource
+	guaranteed := app.GetTag(common.AppTagNamespaceResourceGuaranteed)
+	if quota == "" && guaranteed == "" {
```

Comment:
Is guaranteed mandatory? or Use only when annotation has it?

Reply:
No @manirajv06 , it only uses when user annotation it just like quota, every annotation can be used independently.

Reply:
If we can treat each annotation independently, then we should not put together in a single if checks. We can have separate if checks for each one.